src/eq_fault_geom/geomio/cfm_shp_to_vtk.py

```diff
@@ -0,0 +1,214 @@
+    (xl, yl) = line.xy
+    xt = np.array(xl)
+    yt = np.array(yl)
+    num_trace_points = xt.shape[0]
+    num_points = 2*num_trace_points
+    pt = np.column_stack((xt, yt, np.zeros_like(xt)))
+
+    # Get distance along dip vector.
+    elev_max = -1000.0*fault_info[depth_use]
+    if (np.abs(elev_max) < eps):
+        elev_max = default_elev
+    if (dip != 0.0):
+        dist = elev_max/dip_vec[2]
+    else:
+        dist = default_length
+
+    # Create points at depth.
+    pd = pt + dist*dip_vec
+    points = np.concatenate((pt, pd), axis=0)
+
+    # Create connectivity.
+    if (cell_type == "quad"):
+        cells = create_quads(num_trace_points)
+    else:
+        cells = create_triangles(num_trace_points)
+
+    # Create mesh using meshio.
+    mesh = meshio.Mesh(points, cells)
+
+    return mesh
+
+
+def create_quads(num_trace_points):
+    """
+    Create quad cells given the number of points at the surface.
+    """
+    num_cells = num_trace_points - 1
+    cell_array = np.zeros((num_cells, 4), dtype=np.int)
+    for cell_num in range(num_cells):
+        cell_array[cell_num,0] = cell_num
+        cell_array[cell_num,1] = cell_num + 1
+        cell_array[cell_num,2] = cell_num + num_trace_points + 1
+        cell_array[cell_num,3] = cell_num + num_trace_points
+
+    cells = [("quad", cell_array)]
+
+    return cells
+
+
+def create_triangles(num_trace_points):
+    """
+    Create quad cells given the number of points at the surface.
+    """
+    num_quads = num_trace_points - 1
+    num_triangles = 2*num_quads
+    cell_array = np.zeros((num_triangles, 3), dtype=np.int)
+    cell_num = 0
+    for quad_num in range(num_quads):
+        cell_array[cell_num,0] = quad_num
+        cell_array[cell_num,1] = quad_num + 1
+        cell_array[cell_num,2] = quad_num + num_trace_points + 1
+        cell_num += 1
+        cell_array[cell_num,0] = quad_num + num_trace_points + 1
+        cell_array[cell_num,1] = quad_num + num_trace_points
+        cell_array[cell_num,2] = quad_num
+        cell_num += 1
+
+    cells = [("triangle", cell_array)]
+
+    return cells
+    
+    
+def create_stirling_vtk(fault_info: pd.Series, section_id: int, nztm_geometry: LineString):
+    """
+    Create 3D Stirling fault file from 2D map info.
+    """
+    # Get dip rotation matrix and create mesh from surface info.
+    dip_dir = fault_info["Dip_Dir"]
+    dip_rotation = calculate_dip_rotation(nztm_geometry, dip_dir)
+    mesh = create_mesh_from_trace(fault_info, nztm_geometry, dip_rotation)
+
+    # Write mesh.
+    file_name = fault_info["FZ_Name"].replace(" ", "_")
+    # Note this only works for faults numbered 1-9.
+    if (file_name[-1].isnumeric()):
+        file_list = list(file_name)
+        file_list[-1] = num_replace_dict[file_name[-1]]
+        file_name = "".join(file_list)
+    file_path = Path(file_name)
+    output_file = Path.joinpath(output_path, file_path).with_suffix(vtk_suffix)
```

Comment:
If this function returns the mesh object then:

 - it's easier to test. 
 - it's much more versatile - the user can choose if/how/where to do their IO.

Reply:
Changed function to return mesh object.